prometheus/exporter/intervalpush/exporter.go

```diff
@@ -39,10 +42,20 @@ func (e *exporter) Export(exitChan <-chan bool) {
 		select {
 		case <-exitChan:
 			return
+		case w := <-e.done:
+			e.t.Stop()
+			w <- e.p.Push()
+			return
 		case <-e.t.C:
 			if err := e.p.Push(); err != nil {
 				log.Noticef("Push to gatherer failed: %s", err.Error())
 			}
 		}
 	}
 }
+
+func (e *exporter) Close() error {
+	w := make(chan error)
+	e.done <- w
+	return <-w
```

Comment:
if Export is not running it will hang forever

Reply:
Fair.